sig-and-wg/sig-core/proposals/objects-store-proposal.md

```diff
@@ -0,0 +1,155 @@
+
+### Bucket custom resource
+
+You might want to use different bucket per solution. This is why you need to be able to specify multiple buckets in the ObjectStore. For example one bucket for documentation and one bucket per Web application.
+
+Another use cases for having multiple Buckets configuration:
+- Future extensibility by alowing bucket policy specification per solution
+- Minio doesn't support setting a bucket to behave as a static website host. Future controller of the Bucket custom resource will have to handle this additional functionality
+- Minio doesn't support setting a CDN for your objects. Future controller of the Bucket custom resource will have to handle this additional functionality. For example controller will be reesponsible for configuring a CloudFront for your bucket on S3
+
+Example resource for first version of the ObjectStore:
+```
+apiVersion: bucket.objectstore.kyma-project.io/v1alpha1
+kind: Bucket
+metadata:
+  name: my-bucket
+  namespace: stage
+#ObjectStore 2.0  
+#spec:
+#  policy: public #or other policies
+status:
+  ready: False
+  reason: BucketCreationFailure
+  message: "service unavailable"
+```
+
+You reference the Bucket in your Object CR with the following info in the spec:
+```
+  bucketRef:
+    name: my-bucket
+```
+
+It must be provided because the Object controller checks the Bucket custom resource status to make sure the bucket exists.
+
+## Object custom resource
+
+Object resource mandatory information is the:
+- reference info about the source file/object location that must be fetched by ObjectStore with 2 different modes:
+  - direct link fetch - the link points directly to file that need to be fetched
+  - index.yaml ref - the link to index file that contains reference to files that need to be separately fetch from a given relative location
+```
+#sample of index file with markdown and assets files
+apiVersion: v1
+files:
+  - name: 01-overview.md
+    metadata:
+      title: MyOverview
+      type: Overview
+  - name: 02-details.md
+    metadata:
+      title: MyDetails
+      type: Details
+  - name: 03-installation.md
+    metadata:
+      title: MyInstallation
+      type: Tutorial
+  - name: assets/diagram.svg
+```
+- reference to the bucket where the object should be stored
+
+
+```
+apiVersion: object.objectstore.kyma-project.io/v1alpha1
+kind: Object
+metadata:
+  name: my-package-objects
+spec:
+  source:
+    package: https://some.domain.com/my.zip
+  bucketRef:
+    name: my-bucket
+---
+apiVersion: objectstore.kyma-project.io/v1alpha1
+kind: Object
+metadata:
+  name: my-indexbased-objects
+spec:
+  source:
+    index: https://some.domain.com/index.yaml
+    rewrites:
+      - regex: 
+          find: \stitle="(.*)?"\s*(/>*)
+          replace: $2<title>$1</title>
+  filesFrom:
+    - configMapRef:
+        name: additional-object
+    - configMapRef:
+        name: one-more-object
+  bucketRef:
+    name: my-bucket
+status:
+  ready: True
```

Comment:
what will the status be if one of the files in the index was not fetched? what will happen with files that were uploaded so far? will they be deleted? or partially available?